neqo-transport/src/cc/classic_cc.rs

```diff
@@ -342,6 +247,100 @@ impl CongestionControl {
         );
     }
 
+    /// Whether a packet can be sent immediately as a result of entering recovery.
+    #[must_use]
+    fn recovery_packet(&self) -> bool {
+        self.state == State::RecoveryStart
+    }
+}
+
+impl ClassicCongestionControl {
+    pub fn new(alg: CongestionControlAlgorithm) -> Self {
```

Comment:
If you template based on the algorithm, then you don't need to match anywhere here (the matching happens outside).  You might need a static `::new` function on the trait though, I think.